client/valor/coretypes.py

```diff
@@ -591,7 +588,7 @@ def get_datums(
             raise ValueError(
                 "Cannot filter by dataset_names when calling `Dataset.get_datums`."
             )
-        filter_["dataset_names"] = [self.get_name()]
+        filter_["dataset_names"] = [self.name]  # type: ignore
```

Comment:
Adding type ignores is why i had avoided this previously. This will affect IDE autocompletion for users that have pylance enabled.

Reply:
yea i think this is worth the tradeoff though